java/runtime/src/main/java/io/ray/runtime/util/ConcurrencyGroupUtils.java

```diff
@@ -0,0 +1,90 @@
+package io.ray.runtime.util;
+
+import com.google.common.base.Preconditions;
+import io.ray.api.Ray;
+import io.ray.api.concurrencygroup.ConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroups;
+import io.ray.api.concurrencygroup.annotations.UseConcurrencyGroup;
+import io.ray.api.function.RayFuncR;
+import io.ray.runtime.ConcurrencyGroupImpl;
+import io.ray.runtime.RayRuntimeInternal;
+import io.ray.runtime.functionmanager.JavaFunctionDescriptor;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/// TODO: cache this.
+public final class ConcurrencyGroupUtils {
+
+  public static List<ConcurrencyGroup> extractConcurrencyGroupsByAnnotations(
+      RayFuncR<?> actorConstructorLambda) {
+    SerializedLambda serializedLambda = LambdaUtils.getSerializedLambda(actorConstructorLambda);
+    Class<?> actorClz = getReturnTypeFromSignature(serializedLambda.getInstantiatedMethodType());
+
+    ArrayList<ConcurrencyGroup> ret = new ArrayList<ConcurrencyGroup>();
+    DefConcurrencyGroups concurrencyGroupsDefinitionAnnotation =
+        actorClz.getAnnotation(DefConcurrencyGroups.class);
+    if (concurrencyGroupsDefinitionAnnotation == null) {
+      /// This actor is not defined with concurrency groups definition.
+      return ret;
```

Comment:
Should we throw an error if `DefConcurrencyGroup` is not used but `UseConcurrencyGroup` is used?

Reply:
Sounds good. Will add it.

Reply:
Seems we should allow this case? Because definition is able to be specified dynamically.

Reply:
That makes sense. Will a user-friendly error message be printed if a user tries to use a concurrent group which is not defined?

Reply:
Good idea. I'm following it at #21644 , and I'm merging this PR.